lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
The extra brackets are redundant. The example you point at is already a character group. You don't need an extra set of brackets if you're already in one when the objective is to add more characters to the same character group.
The appointed ticket seems somewhat unrelated.